pkg/kubelet/images/image_gc_manager.go

```diff
@@ -338,13 +338,29 @@ func (im *realImageGCManager) freeSpace(bytesToFree int64, freeTime time.Time) (
 	im.imageRecordsLock.Lock()
 	defer im.imageRecordsLock.Unlock()
 
+	// Make the ListImages into a map to grab an image by ID
+	allImages, err := im.runtime.ListImages()
+	if err != nil {
+		return 0, err
+	}
+	imagesMap := make(map[string]container.Image, len(allImages))
+	for _, img := range allImages {
+		imagesMap[img.ID] = img
+	}
+
```

Comment:
After reading the code, I think another way is adding pinned for imageRecords and set it in `func detectImages()`. 

There is an iterator in detectImages. We can iterate once. Like image.size. 
https://github.com/kubernetes/kubernetes/blob/10976cbb43022c01a7ece07079cb9cedf04aeb90/pkg/kubelet/images/image_gc_manager.go#L240-L260